packages/cadl-vscode/src/extension.ts

```diff
@@ -71,7 +61,18 @@ function resolveCadlServer(context: ExtensionContext): Executable {
     };
   }
 
-  return { command, args, options };
+  // Default to cadl-server on PATH, which would come from `npm install -g
+  // @cadl-lang/compiler` in a vanilla setup.
+  if (!serverPath) {
+    const executable = process.platform === "win32" ? "cadl-server.cmd" : "cadl-server";
+    return { command: executable, args, options };
+  }
+
+  if (!serverPath.endsWith(".js")) {
+    serverPath = join(serverPath, "cmd/cadl-server.js");
```

Comment:
I wonder if we can make this support all cases and stay compatible. 

If ends with .js, run with node
If not ends with .js and (path)/cmd/cadl-server.js exists, use that.
Else run as executable

In addition to being non-breaking, it gives you an escape hatch if you need to run with a node not on path or whatever, you could always craft your own executable.

Not really sure this is necessary, but thought I'd throw it out there.